controllers/clusterversion_controller.go

```diff
@@ -436,3 +447,41 @@ func (c *ClusterVersionReconciler) ensureConsolePlugin() error {
 
 	return nil
 }
+
+func (c *ClusterVersionReconciler) getDeployCSIConfig() (bool, error) {
+	operatorConfig := &corev1.ConfigMap{}
+	operatorConfig.Name = operatorConfigMapName
+	operatorConfig.Namespace = c.OperatorNamespace
+	if err := c.get(operatorConfig); err != nil {
+		return false, fmt.Errorf("failed to get operator configmap: %v", err)
+	}
+
+	data := operatorConfig.Data
+	if data == nil {
+		data = map[string]string{}
+	}
+
+	if value, ok := data[deployCSIKey]; ok {
+		deployCSI, err := strconv.ParseBool(value)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse value for %q in operator configmap as a boolean: %v", deployCSIKey, err)
+		}
+		return deployCSI, nil
+	}
+
+	// verify existence of storagecluster crd as configmap didn't mention deploying CSI
+	storageClusterCRD := &metav1.PartialObjectMetadata{}
+	storageClusterCRD.SetGroupVersionKind(
+		extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"),
+	)
+	storageClusterCRD.Name = "storageclusters.ocs.openshift.io"
+	err := c.get(storageClusterCRD)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return false, fmt.Errorf("failed to verify existence of storagecluster crd: %v", err)
+	}
+	return kerrors.IsNotFound(err), nil
```

Comment:
```suggestion

	var deployCSI bool 
	if value, ok := data[deployCSIKey]; ok {
		if deployCSI, err := strconv.ParseBool(value); err != nil {
			return false, fmt.Errorf("failed to parse value for %q in operator configmap as a boolean: %v", deployCSIKey, err)
		}
	} else {
	        // CSI installation is not specified explicitly in the configmap, behaviour is different in case we
	        // recognize the StorageCluster API on the cluster. 
		storageClusterCRD := &metav1.PartialObjectMetadata{}
		storageClusterCRD.Name = "storageclusters.ocs.openshift.io"
		storageClusterCRD.SetGroupVersionKind(
			extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"),
		)
		if err := c.get(storageClusterCRD); err != nil  {
			if !kerrors.IsNotFound(err) {
				return false, fmt.Errorf("failed to verify existence of storagecluster crd: %v", err)
			}
			deployCSI = false
		} else {
			deployCSI = true
		}
	 }
	return DeployCSI, nil
```

Reply:
done (similar to above).